test/oauth2_unittest.cc

```diff
@@ -118,6 +122,54 @@ TEST_F(OAuth2Test, GetAuthHeaderValueUsingTokenFromMetadataServerAsFallback) {
   EXPECT_EQ("Bearer the-access-token", auth.GetAuthHeaderValue());
 }
 
+TEST_F(OAuth2Test, PropagateGceApiRequestErrorsCumulativeToView) {
+  testing::FakeServer oauth_server;
+  testing::TemporaryFile credentials_file(
+    std::string(test_info_->name()) + "_creds.json",
+    "{\"client_email\":\"user@example.com\",\"private_key\":\"some_key\"}");
+  Configuration config(std::istringstream(
+      "CredentialsFile: '" + credentials_file.FullPath().native() + "'\n"
+  ));
+  Environment environment(config);
+  OAuth2 auth(environment);
+  SetTokenEndpointForTest(&auth, oauth_server.GetUrl() + "/oauth2/v3/token");
+
+  // Remove existing view on kGceApiRequestErrors to avoid other tests
+  // affecting the result, then flush to propagate existing records
+  // to views, including records from other tests.
+  ::opencensus::stats::StatsExporter::RemoveView(
+      ::google::Metrics::kGceApiRequestErrors);
+  ::opencensus::stats::testing::TestUtils::Flush();
+
+  // We want to test this auth.GetAuthHeaderValue() that it records and
+  // register the view, but as inside the auth.GetAuthHeaderValue() we
+  // use singleton pattern, once we remove the view in previous section, it is
+  // unable to add the view back within auth.GetAuthHeaderValue(). Add a view
+  // to export here helps us to test the data inside the view, it also makes
+  // the test fragile if auth.GetAuthHeaderValue() does not register view
+  // correctly.
+  // TODO: Fix this hack if we can contribute to opencensus that when we add
+  // a view, we only add view if a view name is not existed. Current
+  // StatsExporter::AddView always add a new view and destroy the old one,
+  // which is not suitable for our
+  // Metrics::GceApiRequestErrorsCumulativeViewDescriptor().
+  ::google::Metrics::GceApiRequestErrorsCumulativeViewDescriptor().
+      RegisterForExport();
+  auth.GetAuthHeaderValue();
+  // Flush to propagate existing records to views.
+  ::opencensus::stats::testing::TestUtils::Flush();
+
+  EXPECT_THAT(::opencensus::stats::StatsExporter::GetViewData(),
+              ::testing::Contains(::testing::Pair(
+                      Metrics::GceApiRequestErrorsCumulativeViewDescriptor(),
+                      ::testing::Property(
+                          &::opencensus::stats::ViewData::int_data,
```

Comment:
Can we pull this out into `Metrics::GetInt64Data(const std::string& name)`? That way this test could become:
```c++
  EXPECT_THAT(Metrics::GetInt64Data(Metrics::kGceApiRequestErrors),
              ::testing::UnorderedElementsAre(
                ::testing::Pair(::testing::ElementsAre("oauth2"), 1)));
```